src/python/pants/backend/python/target_types.py

```diff
@@ -545,6 +564,19 @@ class PexEmitWarningsField(TriBoolField):
     def value_or_global_default(self, pex_binary_defaults: PexBinaryDefaults) -> bool:
         if self.value is None:
             return pex_binary_defaults.emit_warnings
+        warn_or_error(
+            "2.21.0.dev0",
```

Comment:
This is a relatively small amount of code, so maybe we could extend the deprecation for longer, to be nicer to users/make it easier to upgrade (e.g. if someone is using 2.19.0, and then testing major features/fixes in a hypothetical 2.20.0rc3 and 2.21.0.dev4, it's nice if they're not completely blocked until they make changes for "minor" stuff like shuffling fields around, which removing this in 2.21.0.dev0 would force.)

```suggestion
            "2.23.0.dev0",
```

Thoughts?

Reply:
Given the fix-instead-of-deprecate option shot per other review feedback.